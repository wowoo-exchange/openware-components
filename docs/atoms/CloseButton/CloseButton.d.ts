import * as React from 'react';
interface CloseButtonProps {
    /**
     * Callback called on button click
     */
    onClick: () => void;
}
declare const CloseButton: React.FunctionComponent<CloseButtonProps>;
export { CloseButton, CloseButtonProps, };
