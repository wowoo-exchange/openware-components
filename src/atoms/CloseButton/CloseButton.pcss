.cr-close-button {
  position: relative;
  border: none;
  outline: none;
  cursor: pointer;
  padding: calc(var(--gap) * 2);
  background: transparent;

  &::before,
  &::after {
    background: var(--base-2);
    content: '';
    display: block;
    height: calc(var(--gap) * 2.5);
    left: var(--gap);
    position: absolute;
    top: var(--gap);
    width: 1px;
  }

  &::before {
    transform: rotate(45deg);
  }

  &::after {
    transform: rotate(-45deg);
  }
}
