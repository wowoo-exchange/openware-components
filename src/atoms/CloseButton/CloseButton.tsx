import * as React from 'react';

interface CloseButtonProps {
    /**
     * Callback called on button click
     */
    onClick: () => void;
}

const CloseButton: React.FunctionComponent<CloseButtonProps> = (props: CloseButtonProps) => (
    <button className="cr-close-button" onClick={props.onClick} />
);

export {
    CloseButton,
    CloseButtonProps,
};
