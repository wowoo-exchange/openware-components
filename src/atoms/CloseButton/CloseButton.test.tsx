import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import { spy } from 'sinon';
import { CloseButton, CloseButtonProps } from './CloseButton';

const onClickSpy = spy();
const defaultProps: CloseButtonProps = {
    onClick: onClickSpy,
};

const setup = (props: Partial<CloseButtonProps> = {}) =>
    shallow(<CloseButton {...{ ...defaultProps, ...props }} />);

describe('Close Button', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        onClickSpy.resetHistory();
        wrapper = setup({ onClick: onClickSpy });
    });

    it('should render', () => {
       expect(wrapper).toMatchSnapshot();
    });

    it('should call onClick callback', () => {
       wrapper.find('.cr-close-button').simulate('click');
       expect(onClickSpy.calledOnce).toBeTruthy();
    });
});
