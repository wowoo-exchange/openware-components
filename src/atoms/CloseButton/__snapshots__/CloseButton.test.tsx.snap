// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close Button should render 1`] = `
<button
  className="cr-close-button"
  onClick={[Function]}
/>
`;
